Mage/src/main/java/mage/abilities/effects/common/PreventDamageByTargetEffect.java

```diff
@@ -80,4 +81,29 @@ public String getText(Mode mode) {
         }
         return preventText;
     }
+
+    @Override
+    public boolean hasHint() {
+        return true;
+    }
+
+    @Override
+    public String getHint(Permanent permanent, Ability source, Game game) {
+        if (!this.getTargetPointer().getTargets(game, source).contains(permanent.getId())) {
+            return null;
+        }
+
+        String durationText = duration == Duration.EndOfTurn ? " this turn" : ' ' + duration.toString();
+        String targetText = "{this}";
+        String preventText = (amountToPrevent == Integer.MAX_VALUE ? "Prevent all"
+                : "Prevent the next" + amountToPrevent)
+                + (onlyCombat ? " combat damage " : " damage ");
+        if (passiveVoice) {
+            preventText += "that would be dealt"
```

Comment:
I don’t like that duplicating of text gen code. You try to use simple logic: original rule + result (real value).

Reply:
Let me know if what I pushed will work. I agree there is too much duplicated.